facia-press/app/frontpress/FapiFrontPress.scala

```diff
@@ -332,48 +331,13 @@ trait FapiFrontPress extends EmailFrontPress with Logging {
   private def enrichContent(collection: Collection, content: PressedContent, enriched: Option[EnrichedContent])(implicit executionContext: ExecutionContext): Response[EnrichedContent] = {
     val beforeEnrichment = enriched.getOrElse(EnrichedContent.empty)
 
-    val afterEnrichment = for {
-      embedType <- content.properties.embedType if embedType == "json.html" || embedType == "interactive"
-      embedUri <- content.properties.embedUri
-    } yield {
-      val maybeUpdate = (embedType match {
-        case "json.html" =>
-          (wsClient.url(embedUri).get().map { response =>
-            Json.fromJson[EmbedJsonHtml](response.json) match {
-              case JsSuccess(embed, _) =>
-                beforeEnrichment.copy(embedHtml = Some(embed.html))
-              case _ =>
-                log.warn(s"An embed had invalid json format, and won't be pressed. ${content.properties.webTitle} for collection ${collection.id}")
-                beforeEnrichment
-            }
-          }) recover {
-            case _ => {
-              log.warn(s"A request to an embed uri failed, embed won't be pressed. $embedUri for collection ${collection.id}")
-              beforeEnrichment
-            }
-          }
-        case "interactive" =>
-          val capiId = CapiUrl.extractId(embedUri)
-          capiClient.getResponse(ItemQuery(capiId)).map { response =>
-            response.interactive.flatMap { interactive =>
-              interactive.data match {
-                case atom: com.gu.contentatom.thrift.AtomData.Interactive =>
-                  Some(beforeEnrichment.copy(embedHtml = Some(atom.interactive.html), embedCss = Some(atom.interactive.css), embedJs = atom.interactive.mainJS))
-                case _ =>
-                  None
-              }
-            }.getOrElse({
-              val msg = s"Processing of an interactive atom failed, and it won't be pressed. ${embedUri} for collection ${collection.id}"
-              log.warn(msg)
-              throw new RuntimeException(msg)
-            })
-          }
-      })
-
-      Response.Async.Right(maybeUpdate)
+    val maybeUpdate = content.properties.embedType match {
+      case Some("json.html") => Enrichment.enrichSnap(content.properties.embedUri, beforeEnrichment, collection, wsClient)
+      case Some("interactive") => Enrichment.enrichInteractive(content.properties.atomId, beforeEnrichment, collection, capiClient)
+      case _ => Future(beforeEnrichment)
     }
 
-    afterEnrichment.getOrElse(Response.Right(beforeEnrichment))
+    Response.Async.Right(maybeUpdate)
```

Comment:
I think this is okay because we already have the helpers. But I actually prefer inlining this kind of thing for clarity. E.g.

    Response(maybeUpdate.map(scala.Right.apply))

As it is really obvious to the reader what's going on, and not significantly more verbose. Whereas, at first glance I had no idea what `Response.Async.Right(maybeUpdate)` was doing.

Reply:
Yep agreed that the version you posted is clearer. I'm actually not 100% sure I understand what the `scala.Right.apply` is needed for here, maybe we could chat offline about that.